### PR TITLE
[t3] refactored exception handling

### PR DESCRIPTION
**Pull request by howieraem**
1. Whenever a resource is not found in DB, throw an exception and handle accordingly, rather than returning null
2. If user input violates a unique constraint, handle Spring's DataIntegrityViolationException by extracting the constraint violated

### CODE REVIEW

**Review by KenXiong123 — APPROVED** [review]

File: src/main/java/com/lgtm/easymoney/configs/Consts.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.lgtm.easymoney.configs;
+
+import java.util.Map;
+
+public class Consts {
+    public static final String USER_EMAIL_CONSTRAINT = "user_email_uniq_idx";
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/configs/Consts.java:6]: nice:) it's good to put constants in one place. 
seems like this consts is currently only used for checking db constraints, consider renaming to make it more specific? if this later would be reused to store other constants, add more comments to document its usage

cc: @jenniferduan45  @howieraem  @KenXiong123

File: src/main/java/com/lgtm/easymoney/controllers/UserController.java [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
-import java.util.Arrays;
 
 @RestController
 @RequestMapping("/user")
 public class UserController {
+    private final UserService userService;
+
     @Autowired
-    private UserService userService;
+    public UserController(UserService userService) {
```
Comment (rl3250) [inline src/main/java/com/lgtm/easymoney/controllers/UserController.java:18]: curious: why we can't simply use the original one?
` @Autowired
    private UserService userService;`

**Review by rl3250 — CHANGES_REQUESTED** [review]
commented

File: src/main/java/com/lgtm/easymoney/controllers/UserController.java (hunk 2 above)
Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/controllers/UserController.java:18]: Intellij showed a warning like direct injection is not recommended

File: src/main/java/com/lgtm/easymoney/configs/Consts.java (hunk 1 above)
Comment (howieraem) [reply src/main/java/com/lgtm/easymoney/configs/Consts.java]: You mean rename the class or?

**Review by rl3250 — APPROVED** [review]

**Pull request merged**